phac-nml/irida-next
# Samples table enhancements

## What does this PR do and why?
_Describe in detail what your merge request does and why._

This PR updates the Project and Group Samples table to preserve scroll when sorting, searching, or paging. This also updates the column header names to use `Sample ID` and `Sample Name` for the first two columns instead of `ID` and `Name`. In addition horizontal scrolling of  the table does not result in column headers overlapping with the sticky columns.

## Screenshots or screen recordings
_Screenshots are required for UI changes, and strongly recommended for all other pull requests._

![image](https://github.com/user-attachments/assets/bbafb3a6-616b-478f-abc3-f0f7b7cfb087)

## How to set up and validate locally
_Numbered steps to set up and validate the change are strongly suggested._

1. Run `bundle` to make sure your gems match the ones set in this PR
2. Launch the server
3. Login as any user
4. Navigate to a Project Samples page. Verify that clicking any of the sort links or pagination links preserve scroll location on page. And ensure that all samples actions on that page respond correctly and update the table accordingly.
5. Navigate to a Group Samples page. Verify that clicking any of the sort links or pagination links preserve scroll location on page. And ensure that all samples actions on that page respond correctly and update the table accordingly.

## PR acceptance checklist
This checklist encourages us to confirm any changes have been analyzed to reduce risks in quality, performance, reliability, security, and maintainability.
